src/lib/utils/date_time_utils.cpp

```diff
@@ -4,26 +4,20 @@
 
 namespace opossum {
 
-std::optional<boost::gregorian::date> string_to_date(const std::string& date_string) {
-  // We catch parsing exceptions since we return a std::nullopt if the input string is not a valid date.
-  try {
-    const auto date = boost::gregorian::from_string(date_string);
-    if (!date.is_not_a_date()) {
-      return date;
-    }
-  } catch (const boost::wrapexcept<boost::gregorian::bad_day_of_month>& /* exception */) {
-  } catch (const boost::wrapexcept<boost::gregorian::bad_month>& /* exception */) {
-  } catch (const boost::wrapexcept<boost::gregorian::bad_year>& /* exception */) {
-  } catch (const boost::wrapexcept<boost::bad_lexical_cast>& /* exception */) {}
-  return std::nullopt;
-}
-
-std::optional<boost::posix_time::ptime> string_to_date_time(const std::string& date_time_string) {
+std::optional<boost::posix_time::ptime> string_to_timestamp(const std::string& timestamp_string) {
   // We catch parsing exceptions since we return a std::nullopt if the input string is not a valid timestamp.
   try {
-    const auto date_time = boost::posix_time::time_from_string(date_time_string);
-    if (!date_time.is_not_a_date_time()) {
-      return date_time;
+    if (timestamp_string.size() == 10) {
+      // This is a date without time information.
+      const auto date = boost::gregorian::from_string(timestamp_string);
+      if (!date.is_not_a_date()) {
+        return boost::posix_time::ptime{date};
+      }
+    } else {
+      const auto date_time = boost::posix_time::time_from_string(timestamp_string);
```

Comment:
Since the signature was changed to `timestamp`, do we keep the naming consistent and use `timestamp` here as well?